tests/L0/run_transformer/test_fused_softmax.py

```diff
@@ -216,5 +224,171 @@ def test_autocast_fused_upper_triangle_mask_softmax(self):
                 actual.backward(g0)
                 expected.backward(g1)
 
+
+class TestGenericFusedSoftmaxKernel(common_utils.TestCase):
+
+    def test_forward(self):
+        import generic_scaled_masked_softmax_cuda
+
+        batch = 2
+        attn = 16
+        qlen = 2348
+        klen = 3123
+        scale_t = torch.tensor([1.0])
+        for qlen in [2348, 2322, 1234, 1, 2]:
+            for klen in [
+                3123,
+                1234,
+                2,
+                4,
+                8,
+                3,
+                1,
+                5,
+                10,
+                11,
+                13,
+                128,
+                256,
+                1200,
+                2048,
+                4096,
+                7234,
+                8192,
+                10232,
+                4128,
+            ]:
+                inputs = torch.normal(0, 2, (batch, attn, qlen, klen), dtype=torch.float16, device='cuda:0')
+                masks = torch.randint(0, 2, (batch, 1, qlen, klen), dtype=torch.bool, device='cuda:0')
+                softmax_results = generic_scaled_masked_softmax_cuda.forward(inputs, masks, scale_t[0].item())
```

Comment:
out of curiosity, why do you have a scale in Tensor and then extract and pass a Python scalar from `scale_t` to this function?

Reply:
yeah. passing a python scalar works. changed.